src/toil/resource.py

```diff
@@ -394,7 +394,9 @@ def forModule(cls, name):
                 assert dirPathTail == package
             dirPath = os.path.sep.join(filePath)
         log.debug("Module dir is %s", dirPath)
-        assert os.path.isdir(dirPath)
+        require(os.path.isdir(dirPath),
+                'Bad directory path %s for module %s. Note that hot-deployment does not support \
+                .egg-link files yet.', dirPath, name)
```

Comment:
Can you add in something along the lines of "or scripts located in the root directory" so users who run into #1455 will have some idea whats going on? Keep the ticket open though so we can look at a more permanent solution later.